src/Val.h

```diff
@@ -1403,8 +1423,10 @@ class RecordVal final : public Val, public notifier::detail::Modifiable
 	static void DoneParsing();
 
 protected:
+	friend class zeek::logging::Manager;
```

Comment:
I'm a bit hesitant here, because users of `RawOptField` are responsible for memory management, which is easy to get wrong.  So I'd rather there be a high bar for access to it.

`VectorVal::RawVec` is public just because there are numerous points where ZAM needs access to it, and also potentially `-O C++` code (not yet implemented).

Reply:
That is fair and I agree. I now remember there's also the GetFieldAs() API (and for vectors the type specific At() methods), so I can try to leverage these first.